docs/source/traitsui_user_manual/scripts/regenerate_example_screenshots.py

```diff
@@ -0,0 +1,151 @@
+# (C) Copyright 2004-2021 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+import contextlib
+import os
+from unittest import mock
+from functools import partialmethod
+
+import pkg_resources
+
+from traits.api import HasTraits
+from traitsui.testing.api import UITester
+
+# Demo files are part of the package data.
+DEMO = pkg_resources.resource_filename("traitsui", "examples/demo")
+
+
+def _is_python_file(path):
+    """ Return true if the given path is (public) non-test Python file."""
+    _, basename = os.path.split(path)
+    _, ext = os.path.splitext(basename)
+    return (
+        ext == ".py"
+        and not basename.startswith("_")
+        and not basename.startswith("test_")
+    )
+
+
+def get_python_files(directory):
+    """ Report Python files to be tested or to be skipped.
+
+    Returns
+    -------
+    accepted_files : list of str
+        Python file paths to be tested.
+    """
+    accepted_files = []
+    for root, _, files in os.walk(directory):
+        for filename in files:
+            path = os.path.abspath(os.path.join(root, filename))
+            if not _is_python_file(path):
+                continue
+            accepted_files.append(path)
+
+    # Avoid arbitrary ordering from the OS
+    return sorted(accepted_files)
+
+
+def get_editor_example_filees(python_files):
+    Editor_demo_files = []
+    for filename in python_files:
+        if filename.endswith('Editor_demo.py'):
```

Comment:
did you take a look at the examples to ensure that this convention is followed by all of the relevant examples files?

Reply:
This is true for all the examples in `examples/demo/StandardEditors`.  There are a couple others which endup having this name eg `examples/demo/Advanced/Dynamic_EnumEditor_demo.py` which may not yet be currently included in the docs page.  Others which do _not_ have this name may potentially be relevant in the docs, but are not currently included.  Most of those that do not have this naming convention are showcasing more advanced/specific features and aren't what we would want to screenshot to show in the docs most likely.

This does tie into the need for special handling certain files though, and again this can be done in follow up PR(s) where this script progressively gets more an more complex or other scripts are added